envoy/adapter/adapter.go

```diff
@@ -0,0 +1,307 @@
+package adapter
```

Comment:
There is still some duplication in the functions here and what remained in `sidecarhttp/envoy_api.go`, but I think deduplicating it will be ugly and I'm thinking that it's probably safe to delete that REST API soon, given that Envoy has a 1-2 year [API lifecycle](https://github.com/envoyproxy/envoy/blob/786544375adadb43375eb7783370fde4c3c4b0a6/api/API_VERSIONING.md#api-lifecycle) and the V2 API that I just implemented [will be deprecated](https://www.envoyproxy.io/docs/envoy/latest/api/api_supported_versions) at the end of this year.

I wanted to switch directly to V3, but the `go-control-plane` doesn't seem to have made the necessary updates yet. It should be a much easier transition, though, since V3 is mostly similar to V2.